pkg/cloud/gcp/gcp.go

```diff
@@ -0,0 +1,154 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package gcp provides common functionality to run cloud prepare/cleanup on GCP Clusters.
+package gcp
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/admiral/pkg/reporter"
+	"github.com/submariner-io/admiral/pkg/util"
+	"github.com/submariner-io/cloud-prepare/pkg/api"
+	"github.com/submariner-io/cloud-prepare/pkg/gcp"
+	gcpClientIface "github.com/submariner-io/cloud-prepare/pkg/gcp/client"
+	"github.com/submariner-io/cloud-prepare/pkg/k8s"
+	"github.com/submariner-io/cloud-prepare/pkg/ocp"
+	"github.com/submariner-io/submariner-operator/internal/restconfig"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/dns/v1"
+	"google.golang.org/api/option"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+)
+
+type Config struct {
+	DedicatedGateway bool
+	Gateways         int
+	InfraID          string
+	Region           string
+	ProjectID        string
+	CredentialsFile  string
+	OcpMetadataFile  string
+	GWInstanceType   string
+}
+
+// RunOn runs the given function on GCP, supplying it with a cloud instance connected to GCP and a reporter that writes to CLI.
+// The functions makes sure that infraID and region are specified, and extracts the credentials from a secret in order to connect to GCP.
+func RunOn(restConfigProducer restconfig.Producer, config *Config, creds *google.Credentials, status reporter.Interface,
```

Comment:
```suggestion
func RunOn(restConfigProducer *restconfig.Producer, config *Config, creds *google.Credentials, status reporter.Interface,
```

Reply:
I tried to keep the original function parameters similar.